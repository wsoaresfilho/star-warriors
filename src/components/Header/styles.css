.header {
  background-color: var(--dark-bg-color);
  height: 90px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  color: white;
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
}

.dark .header {
  background-color: var(--light-bg-color);
}

.header__container {
  max-width: var(--container-max-width);
  margin: 0 auto;
}

.header__logo {
  max-height: 130px;
}
