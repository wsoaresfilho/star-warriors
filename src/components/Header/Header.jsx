import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

const Header = props => {
    const { logo } = props;
    return (
        <header className='header'>
            <div className='header__container'>
                <a href='/'>
                    <img className='header__logo' alt='Logo' src={logo} />
                </a>
            </div>
        </header>
    );
};

Header.propTypes = {
    logo: PropTypes.string.isRequired,
};

export default Header;
