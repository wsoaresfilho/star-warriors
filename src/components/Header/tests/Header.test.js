import React from 'react';
import { shallow } from 'enzyme';
import Header from '../Header';

describe('Header wrapper', () => {
    let wrapper;

    beforeAll(() => {
        wrapper = shallow(<Header logo='light' />);
    });

    describe('should render', () => {
        it('header element', () => {
            expect(wrapper.find('.header').length).toBe(1);
        });

        it('header__container element', () => {
            expect(wrapper.find('.header__container').length).toBe(1);
        });

        it('logo link with image', () => {
            expect(wrapper.find('a img.header__logo').length).toBe(1);
        });

        it('light logo image when using light theme', () => {
            expect(wrapper.find('img.header__logo').prop('src')).toBe('light');
        });

        it('dark logo image when using dark theme', () => {
            wrapper.setProps({ logo: 'dark' });
            expect(wrapper.find('img.header__logo').prop('src')).toBe('dark');
        });

        it('snapshot correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});
