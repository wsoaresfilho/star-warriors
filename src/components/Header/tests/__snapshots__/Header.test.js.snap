// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header wrapper should render snapshot correctly 1`] = `ShallowWrapper {}`;
